src/commonMain/kotlin/uk/gov/hmrc/helptosavecalculator/FirstBonusTermCalculator.kt

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.gov.hmrc.helptosavecalculator
+
+import com.soywiz.klock.DateFormat
+import com.soywiz.klock.DateTime
+import com.soywiz.klock.parse
+import uk.gov.hmrc.helptosavecalculator.exceptions.InvalidRegularPaymentException
+import uk.gov.hmrc.helptosavecalculator.models.FirstBonusCalculatorResponse
+import uk.gov.hmrc.helptosavecalculator.utils.monthsSince
+import uk.gov.hmrc.helptosavecalculator.validation.RegularPaymentValidators
+
+object FirstBonusTermCalculator {
+
+    fun runFirstBonusCalculator(
+            regularPayment: Double,
+            currentBalance: Double,
+            paidInThisMonth: Double,
+            accountStartDate: String,
+            firstTermEndDate: String,
+            secondTermEndDate: String,
+            BalanceMustBeMoreThanForBonus: Double
+    ): FirstBonusCalculatorResponse {
```

Comment:
Would this not be better as a data class / object rather than listing each value out?